tests/unit/test_semver.py

```diff
@@ -0,0 +1,210 @@
+from repo2docker.buildpacks.julia import semver
+
+
+def test_str_to_version():
+    assert semver.str_to_version("1.5.2") == (1, 5, 2)
+    assert semver.str_to_version("1") == (1,)
+
+
+def test_major_minor_patch():
+    V = (1, 2, 3)
+    assert (semver.major(V), semver.minor(V), semver.patch(V)) == (1, 2, 3)
+    assert semver.major((1,)) == 1
+    assert semver.minor((1,)) == 0
+    assert semver.patch((1,)) == 0
+
+
+def test_simple_matches():
```

Comment:
In a future refactor, we should use pytest's parameterize capability to feed in versions to get closer to DRY. [See example from JupyterHub tests](https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/tests/test_proxy.py#L169).

Reply:
Good idea -> #615